### PR TITLE
refactor: Show Client Charge

### PR DESCRIPTION
**Pull request by sosnickm**
refactor: Show Client Charge

### CODE REVIEW

**Comment by therajanmaurya** [comment]
@sosnickm There are two problems with the PR and it can't be merged until they are fixed.
1. The Build is not passing the travis check.
2. You have updated gradle wrapper and versions, these changes won't be accepted as a part of this PR. They should be sent in a Separate PR or not sent at all.

So please revert you gradle wrapper and build.gradle changes and force push again.

I am sure you travis build will pass :)

**Comment by sosnickm** [comment]
Is there a way that I should do the travis check myself? I believe that I changed the gradle back. If not, please let me know what to do.

**Comment by therajanmaurya** [comment]
Your travis build is failing because of Checkstyle rule violations.
Checkstyle rule violations were found.  
```See the report at: directory/self-service-app/app/build/reports/checkstyle/checkstyle.html```

**Comment by sosnickm** [comment]
I am trying to find the bug report, but the link it is giving me at file:///home/travis/build/openMF/self-service-app/app/build/reports/findbugs/findbugs.html does not work. Do you know what the correct link is?

**Comment by therajanmaurya** [comment]
before pushing code run this command in android studio terminal 
```gradlew check``` and If any error will occur then look to the log and here is the report directory ```ProjectDirectory/self-service-app/app/build/reports```

**Comment by sosnickm** [comment]
After running gradlew check it says "bash: gradlew: command not found" should I download something to fix this?

**Comment by therajanmaurya** [comment]
Sorry, the command is ```./gradlew check```

File: app/src/main/java/org/mifos/selfserviceapp/models/Charge.java [hunk 1]
```diff
@@ -63,9 +118,62 @@ public void setCurrency(Currency currency) {
...
+
+    public void setWaived(boolean isWaived) {
+        this.isWaived = isWaived;
+    }
+
+
     public static class Currency {
         private String code;
         private String name;
+        private int decimalPlaces;
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/models/Charge.java]: You have created this POJO inside the ```Charge``` POJO that is not the good idea, make it separate

File: app/src/main/java/org/mifos/selfserviceapp/models/Charge.java [hunk 2]
```diff
@@ -82,6 +190,100 @@ public String getName() {
...
+        }
+
+        public void setDisplayLabel(String displayLabel) {
+            this.displayLabel = displayLabel;
+        }
+
+    }
+
+
+    public static class ChargeTimeType {
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/models/Charge.java]: You have created this POJO inside the ```Charge``` POJO that is not the good idea, make it separate

File: app/src/main/java/org/mifos/selfserviceapp/models/Charge.java [hunk 3]
```diff
@@ -82,6 +190,100 @@ public String getName() {
...
+            return value;
+        }
+
+        public void setValue(String value) {
+            this.value = value;
+        }
     }
 
+
+    public static class ChargeCalculationType {
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/models/Charge.java]: You have created this POJO inside the ```Charge``` POJO that is not the good idea, make it separate

File: app/src/main/res/drawable/circle_on.xml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (therajanmaurya) [inline app/src/main/res/drawable/circle_on.xml]: We have already implemented the circle view for showing the status So remove this and 
follow this PR https://github.com/openMF/self-service-app/pull/53

File: app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java [hunk 5]
```diff
@@ -46,17 +49,65 @@ public Charge getItem(int position) {
...
+    // Also changes the color of the circle depending on whether the charge is active or not
     @Override
     public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
         if (holder instanceof RecyclerView.ViewHolder) {
 
             Charge charge = getItem(position);
-            ((ClientChargeAdapter.ViewHolder) holder).tvAmount.setText(
-                    String.valueOf(charge.getAmount() + charge.getCurrency().getCode()));
+            Currency c  = Currency.getInstance("" + (charge.getCurrency()).getCode());
+            ((ClientChargeAdapter.ViewHolder) holder).tvAmountDue.setText(
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java:56]: Why are you showing the ```Currency``` code, there is no need for this. you can easily get the ```amount``` by calling ```charge.getAmount()``` and same for others below.

File: app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java [hunk 6]
```diff
@@ -46,17 +49,65 @@ public Charge getItem(int position) {
...
+            GradientDrawable bgShape = (GradientDrawable) iv.getDrawable();
+            if (charge.isIsActive()) {
+                bgShape.setColor(Color.rgb(129, 209, 53));
+            } else {
+                bgShape.setColor(Color.rgb(255, 255, 255));
+            }
+
+            String day = charge.getDueDate().get(2).toString() + " ";
+            String month = "";
+            switch (charge.getDueDate().get(1).toString()) {
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java]: This is not a good way to get the month name. 
Look Here : http://stackoverflow.com/a/5799175/3176277 Implement this

File: app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java (hunk 5 above)
Comment (sosnickm) [reply app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java:56]: I am doing this here not to get the amount but to get the symbol i.e. $

Comment (sosnickm) [reply app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java:56]: If you want me to change it, then let me know what you would like to see

Comment (therajanmaurya) [reply app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java:56]: Ok That's fine :)

**Comment by therajanmaurya** [comment]
can you please post the screenshot of your changes.

**Comment by sosnickm** [comment]
screen shot 2016-12-12 at 1 05 22 am

**Comment by therajanmaurya** [comment]
remove the ```Client Name```, stay with only charge name like example ```dscs```

File: app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java [hunk 7]
```diff
@@ -46,21 +49,52 @@ public Charge getItem(int position) {
...
+            } else if (charge.isIsPaid()) {
+                ((ViewHolder) holder).circle_status
+                        .setImageDrawable(setCircularBackground(R.color.light_yellow));
+            } else {
+                ((ViewHolder) holder).circle_status
+                        .setImageDrawable(setCircularBackground(R.color.light_blue));
+            }
+
+            String day = charge.getDueDate().get(2).toString();
+            String month = new DateFormatSymbols().getMonths()
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java]: Make a Utils.java class in /utils directory and add a method ```getMonth(int month)``` and put this logic there and call here.

File: app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java [hunk 8]
```diff
@@ -46,21 +49,52 @@ public Charge getItem(int position) {
...
+                            [Integer.parseInt(charge.getDueDate().get(1).toString()) - 1];
+            String year = charge.getDueDate().get(0).toString();
+
+            ((ViewHolder) holder).tvDueDate.setText(day + " " + month + " " + year);
         }
 
     }
 
+    private LayerDrawable setCircularBackground(int colorId) {
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java]: I found this method is using multiple places, can you please put this also in Utils.java and call here.

Comment (sosnickm) [reply app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java]: Done, I also changed all the other files that call it as well

File: app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java (hunk 7 above)
Comment (sosnickm) [reply app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java]: Done

**Comment by therajanmaurya** [comment]
looks good, squash your commits into one I will merge :)

**Comment by sosnickm** [comment]
Hey rajan, I messed up trying to squash all my commits and ended up recommiting a bunch of things from before. I am unsure how to fix this or what to do, can you help me?

**Pull request closed without merging**